qiskit_experiments/curve_analysis/guess.py

```diff
@@ -34,29 +50,46 @@ def frequency(x: np.ndarray, y: np.ndarray) -> float:
     Args:
         x: Array of x values.
         y: Array of y values.
+        filter_window: Window size of Savitzky-Golay filter. This should be odd number.
+        filter_dim: Dimension of Savitzky-Golay filter.
 
     Returns:
         Frequency estimation of oscillation signal.
     """
-    sampling_rates = np.unique(np.diff(x))
-    sampling_rates = sampling_rates[sampling_rates > 0]
+    sampling_interval = np.unique(np.diff(x))
+    sampling_interval = sampling_interval[sampling_interval > 0]
 
-    if len(sampling_rates) != 1:
-        sampling_rate = np.min(sampling_rates)
-        x_ = np.arange(x[0], x[-1], sampling_rate)
+    if len(sampling_interval) != 1:
+        sampling_interval = np.min(sampling_interval)
+        x_ = np.arange(x[0], x[-1], sampling_interval)
         y_ = np.interp(x_, xp=x, fp=y)
     else:
-        sampling_rate = sampling_rates[0]
+        sampling_interval = sampling_interval[0]
         x_ = x
         y_ = y
 
     fft_data = np.fft.fft(y_ - np.average(y_))
-    freqs = np.fft.fftfreq(len(x_), sampling_rate)
+    freqs = np.fft.fftfreq(len(x_), sampling_interval)
 
     positive_freqs = freqs[freqs >= 0]
     positive_fft_data = fft_data[freqs >= 0]
 
-    return positive_freqs[np.argmax(np.abs(positive_fft_data))]
+    freq_guess = positive_freqs[np.argmax(np.abs(positive_fft_data))]
+
+    if freq_guess < 1 / (sampling_interval * len(x)):
```

Comment:
You're right. I just noticed this problem. Do you have any suggestion of the condition? The most easiest one would be `<=`.

Reply:
Maybe `< 1.5 / (sampling_interval * len(x))`? That is just to avoid any floating point rounding issues if fftfreq calculates the frequency points slightly differently from `1 / (sampling_interval * len(x))`